roomserver/storage/postgres/purge_statements.go

```diff
@@ -0,0 +1,159 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/matrix-org/dendrite/internal/sqlutil"
+	"github.com/matrix-org/dendrite/roomserver/types"
+)
+
+const purgeEventJSONSQL = "" +
+	"DELETE FROM roomserver_event_json WHERE event_nid = ANY(" +
+	"	SELECT event_nid FROM roomserver_events WHERE room_nid = $1" +
```

Comment:
Would a JOIN be a bit more efficient than this? i.e join on the event_nid then delete where the room_nid matches $1.